lib/mosaic-objects/src/main/java/org/eclipse/mosaic/lib/objects/addressing/AdHocMessageRoutingBuilder.java

```diff
@@ -54,8 +72,20 @@ public AdHocMessageRoutingBuilder(String hostName, GeoPoint sourcePosition) {
         );
     }
 
-    private MessageRouting build(DestinationAddressContainer dac) {
-        return new MessageRouting(dac, sourceAddressContainer);
+    public MessageRouting build() {
+        checkNecessaryValues();
+        MessageRouting messageRouting = new MessageRouting(new DestinationAddressContainer(
+                routing, destination, channel, hops, targetArea, ProtocolType.UDP),
+                sourceAddressContainer);
+        resetValues();
```

Comment:
Yes, in some tests the same AdHocMesageRoutingBuilder is used two times. That resulted in failures. If you have other ideas or think we don't need this feel free to propse them. Otherwise I should write tests for this yes.

Reply:
thinking of it now: given that you cannot set destination and routing multiple times, you just have to discard the AdHocMesageRoutingBuilder in your tests, and use a new one for every test case.

Otherwise you could have done something like:

```
resetBuilder(builder b){
b.destination(null)
b.hops(null)
b.routing(null)
... etc ...
} 

b = new Builder()
.... b.testMe() ....
resetBuilder(b)
... b.otherTest()...
```
in your testfile, and thereby reset the values inside the builder

Reply:
I agree with @hoelger that the build should not be re-usable.